specs/modal/readme.md

```diff
@@ -0,0 +1,164 @@
+# Modal
+
+Brief component description
+
+The **Modal** component is a window that opens on top of the application's main window. It blocks the main application's interactivity, and the application flow. In practice, it usually leaves the main app visible in the background, faded out. Users must interact with the modal in order to return to the application flow.
+
+## Elements
+
+![Modal](./assets/modal-basic.png)
+
+The modal consists of a header, body and footer. The header usually contains a title and optionally a close button, the body contains the main modal content, and the footer usually contains a primary button and another close button. The modal has a backdrop that blocks interactivity with the rest of the screen.
+
+Interacting with the modal can consist of:  
+- clicking its primary button
+- clicking its backdrop
+- clicking its close/cancel button(s)
+
+## Use Cases
+
+* A modal can be used to edit a complex formula inside of a spreadsheet
+* A modal can be used in order to require sign-in before using a certain feature of an app
+* A modal can be used to display a photograph or video in focus on top of the screen
+
+## API
+
+### Component Props
+
+| Prop | Type | Default | isRequired | Description |
+| --- | --- | --- | --- | --- |
+| isOpen | bool | false | yes | Managed by state; this is how to control the visibility of the modal|
+| onAfterOpen | func | undefined |  |  |
+| onRequestClose(source) | func | undefined |  | The function used to handle cancel events on the modal; typically sets the open state to false. It is passed a source, as one of a number of possible closing options |
+| children | any |  |  | children nodes of the modal |
+| width | number/enum | 'medium' |  | Explicitly set a numeric width or provide one of three sizes; 'small', 'medium', 'large' - 320px, 640px, 960px respectively|
+| height | number/enum | 'auto' |  | Explicitly set a numeric height of the modal|
+| container | any | null |  |  |
+| aria-* | - | - |  |  |
+| aria-role | - | 'dialog' |  |  |
+
+#### Modal Roles
+
+Modal children can have a `role` attribute that allows them to declare which part of the modal is described by the node.
```

Comment:
Please note that it won't probably be`role` attribute.
1. `role` attr already has different meaning in HTML
2. as a custom attribute it must be `data-*`
3. the hottest candidate under discussion right now is `data-slot` but the discussion is still under way.